### PR TITLE
sheet: Fix scrolling up when a buffer refill is required

### PR DESCRIPTION
**Pull request by richiejp**
There is a bug which happens when scrolling up requires a buffer refill. Essentially it appears to be off-by-one when deciding if the desired row is outside the buffers lower boundary. It can be reproduced by pressing G then C-u or k repeatedly until it gets stuck.

Adding the header_span value to the lower boundary appears to resolve the issue, but I'm not completely sure this is the correct solution.

There's another bug where scrolling down one page from the top then trying to scroll up one page does not work unless the arrow keys are used to go a bit further down first. It doesn't appear to be related to this one.

### CODE REVIEW

**Comment by liquidaty** [comment]
@richiejp thank you! Is there a trivial test we can add, that should pass with this change in place (but fail when this change is not applied)?

**Comment by richiejp** [comment]
I added a test to scroll up from the bottom to the top (it skips a test number due to a conflict with my other PR). Thanks!

**Comment by liquidaty** [comment]
@richiejp great, thank you. It looks like the test requires a file "expected/test-sheet-9.out" that needs to be committed to the repo. Could you add that and then confirm that the CI tests pass?

**Comment by richiejp** [comment]
Sorry that catches me out every time, I have added a .gitignore for that dir so that the .out files are not ignored there.

File: app/test/Makefile [hunk 1]
```diff
@@ -672,3 +672,15 @@ test-sheet-8: ${BUILD_DIR}/bin/zsv_sheet${EXE}
...
 	${CMP} ${TMP_DIR}/$@.out expected/$@.out && ${TEST_PASS} || (echo 'Incorrect output:' && cat ${TMP_DIR}/$@.out && ${TEST_FAIL}))
+
+test-sheet-9: ${BUILD_DIR}/bin/zsv_sheet${EXE}
+	@${TEST_INIT}
+	@echo 'set-option default-terminal "tmux-256color"' > ~/.tmux.conf
+	@(tmux new-session -x 80 -y 6 -d -s $@ "${PREFIX} $< -d 3 ${TEST_DATA_DIR}/test/mixed-line-endings.csv" && \
+	sleep 0.5 && \
+	tmux send-keys -t $@ "G" && \
+	tmux send-keys -t $@ -N 4096 "k" && \
+	sleep 2 && \
```
Comment (iamazeem) [inline app/test/Makefile:683]: You may adjust `-y` (height) above to reduce the number of repeat count `-N` and delay here.

**Pull request merged**